agent/api/transitiondependency.go

```diff
@@ -19,3 +26,35 @@ type ContainerDependency struct {
 	// resource satisfies the dependency
 	DependentStatus ContainerStatus `json:"DependentStatus,omitempty"`
 }
+
+// TransitionDependenciesMap is a map of the dependent container status to other
+// dependencies that must be satisfied.
+type TransitionDependenciesMap map[ContainerStatus]TransitionDependencySet
+
+// UnmarshalJSON decodes the TransitionDependencySet tag in the JSON encoded string
+// into the TransitionDependenciesMap object
+func (td *TransitionDependenciesMap) UnmarshalJSON(b []byte) error {
+	validMap := make(map[ContainerStatus]TransitionDependencySet)
```

Comment:
Won't `make(TransitionDependenciesMap)` work? Also, can you use the name `dependencyMap` or similar?

Reply:
Ideally, `make(TransitionDependenciesMap)` should work. It's weird that I'm running into stack overflow error during unmarshal JSON when I have this as `make(TransitionDependenciesMap)`. thinking if it's due to `ContainerStatus` being present as key in the map as well as in `TransitionDependencySet` (cyclic data?)